src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/XamlSetMarkupExtensionEventArgs.cs

```diff
@@ -9,17 +9,17 @@ namespace System.Windows.Markup
     public class XamlSetMarkupExtensionEventArgs : XamlSetValueEventArgs
     {
         public XamlSetMarkupExtensionEventArgs(XamlMember member,
-            MarkupExtension value, IServiceProvider serviceProvider) :
+            IMarkupExtension value, IServiceProvider serviceProvider) :
             base(member, value)
         {
             ServiceProvider = serviceProvider;
         }
 
-        public MarkupExtension MarkupExtension { get { return Value as MarkupExtension; } }
+        public IMarkupExtension MarkupExtension { get { return Value as IMarkupExtension; } }
```

Comment:
Breaking change.
Perhaps injecting a base class between this class and XamlSetValueEventArgs that adds the interface version, and leave XamlSetMarkupExtensionEventArgs alone (apart from now inheriting from the intermediate class).